### PR TITLE
fix some codes of pilot/config

### PR DESCRIPTION
**Pull request by gy95**
modify some codes of package `pilot/pkg/config`

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/config/kube/crd/conversion.go [hunk 1]
```diff
@@ -102,6 +102,7 @@ func ConvertObject(schema collection.Schema, object IstioObject, domain string)
 	status, err := IstioStatusJSONFromMap(object.GetStatus())
 	if err != nil {
 		log.Errorf("could not get istio status from map %v, err %v", object.GetStatus(), err)
+		return nil, err
```
Comment (adiprerepa) [inline pilot/pkg/config/kube/crd/conversion.go:105]: I don't think we want to completely error out if there is an issue with status...cc @howardjohn

**Review by adiprerepa — COMMENTED** [review]
is this just a general fix? Could you update the description with a more detailed reasoning?

File: pilot/pkg/config/kube/gateway/conversion.go [hunk 2]
```diff
@@ -503,6 +503,9 @@ func buildTCPVirtualService(obj config.Config, gateways []gatewayReference, doma
 			reportError(err)
 			return nil
 		}
+		if len(route) == 0 {
```
Comment (ramaraochavali) [inline pilot/pkg/config/kube/gateway/conversion.go:506]: Can you add a test for this?

File: pilot/pkg/config/kube/crd/conversion.go (hunk 1 above)
Comment (gy95) [reply pilot/pkg/config/kube/crd/conversion.go]: OK，i've reverted it

File: pilot/pkg/config/kube/gateway/testdata/tcp.status.yaml.golden [hunk 3]
```diff
@@ -80,3 +80,12 @@ status:
       name: gateway
       namespace: istio-system
 ---
+apiVersion: networking.x-k8s.io/v1alpha1
+kind: TCPRoute
+metadata:
+  creationTimestamp: null
+  namespace: default
+spec: null
```
Comment (howardjohn) [inline pilot/pkg/config/kube/gateway/testdata/tcp.status.yaml.golden:88]: This doesn't seem legal

File: pilot/pkg/config/kube/gateway/testdata/tcp.status.yaml.golden [hunk 4]
```diff
@@ -80,3 +80,12 @@ status:
...
+status:
```
Comment (hzxuzhonghu) [inline pilot/pkg/config/kube/gateway/testdata/tcp.status.yaml.golden:89]: Is this to test the above change?

I donot think with this config, `buildTCPVirtualService` can run into the clause

Comment (gy95) [reply pilot/pkg/config/kube/gateway/testdata/tcp.status.yaml.golden:89]: i modify this file `pilot/pkg/config/kube/gateway/testdata/tcp.yaml` to trigger it

**Pull request closed without merging**